source/common/network/raw_buffer_socket.cc

```diff
@@ -38,18 +39,20 @@ IoResult RawBufferSocket::doRead(Buffer::Instance& buffer) {
       ENVOY_CONN_LOG(trace, "read error: {}", callbacks_->connection(),
                      result.err_->getErrorDetails());
       if (result.err_->getErrorCode() != Api::IoError::IoErrorCode::Again) {
+        wouldBlock = true;
```

Comment:
I think this is still wrong. Shouldn't we set would_block to true if (result.err_->getErrorCode() == Api::IoError::IoErrorCode::Again)

Reply:
I am trying to set up tests for all the sockets so we can make sure that we got all of them correctly